start_jupyter_cm/linux.py

```diff
@@ -0,0 +1,170 @@
+import os
+import sys
+import stat
+from subprocess import call
+import shutil
+
+from .utils import get_environment_label
+
+# Logic to detect desktop environments on Linux:
+NPATH = os.path.expanduser("~/.local/share/nautilus")
+# User directories might not exist by default in KDE so detect root install:
+KPATH = "/usr/share/kservices5/ServiceMenus"
+no_supported_DE = True
+
+if os.path.isdir(NPATH):
+    GNOME = True
+    no_supported_DE = False
+if os.path.isdir(KPATH):
+    KDE = True
+    no_supported_DE = False
+
+if GNOME:
+    SPATH = os.path.join(NPATH, "scripts")
+
+if KDE:
+    KPATH = os.path.expanduser("~/.local/share/kservices5/ServiceMenus")
+
+if no_supported_DE:
+    raise EnvironmentError('start_jupyter_cm only supports Gnome (+ Nautilus '
+                           'file manager) and KDE (+ Dolphin file manager) '
+                           'desktop environments on Linux, and neither was '
+                           'detected. Nothing has been installed.')
+
+
+# Get python path and anaconda environment:
+PATH = "{}/bin".format(sys.exec_prefix)
+CONDA_ENV_LABEL = get_environment_label()
+
+script = \
+    """#!/usr/bin/python
+
+import sys
+import os.path
+import subprocess
+
+folders = [path for path in sys.argv[1:] if os.path.isdir(path)]
+any_file_selected = len(folders) < len(sys.argv[1:])
+if any_file_selected:
+    subprocess.Popen(["%s/jupyter-%s"])
+for folder in folders:
+    os.chdir(folder)
+    subprocess.Popen(["%s/jupyter-%s"])
+    os.chdir("..")
+
+"""
+
+
+def get_desktop_file(app, logo):
+    """
+    Returns the contents of a desktop file that will launch Jupyter
+    in a certain mode. Required since KDE uses .desktop files instead of
+    just scripts.
+
+    Parameters
+    ----------
+    app : str
+        Either 'notebook', 'qtconsole', or 'lab'
+    logo : str
+        Path to the logo file to use for the .desktop file
+    """
+    desktop_file = \
+        ("\n"
+         "[Desktop Action jupyter-{1}]\n"
+         "Exec='{0}/.local/share/kservices5/ServiceMenus/"
+         "Jupyter_{1}_here{2}' %f\n"
+         "Icon={3}\n"
+         "Name=Jupyter {4}{2} here\n"
```

Comment:
Yup, good call; that was unintentional. I'll update it shortly.

Reply:
This little issue is a good example of why we should embrace f-strings. Why not reverting to using f-string and removing support for 3.5 at this opportunity?

Reply:
:+1: I didn't want to make that decision, but if we're okay removing 3.5, then I'd be happy to swap back to f-strings

Reply:
I am fine with both approaches and I agree that it is not a big deal to drop support for python 3.3, 3.4 and 3.5.

Reply:
Then let's drop support for Python < 3.6. f-string makes the code mode readable, so I think that it'll be good to do the same for the rest of the project in the hyperspy organization.

Reply:
Agree, HyperSpy test suite is currently broken on python 3.5 because of installation issue in conda environment; I have sent a PR with more details to drop python 3.5: https://github.com/hyperspy/hyperspy/pull/2220